godot-macros/src/ffi_macros.rs

```diff
@@ -13,22 +13,34 @@ use crate::util::bail;
 use crate::ParseResult;
 use proc_macro2::TokenStream;
 use quote::{format_ident, quote};
+use std::env;
+use std::sync::atomic::{AtomicU32, Ordering};
+
+// Note: global state in proc-macros may become problematic in the future, see:
+// https://users.rust-lang.org/t/simple-state-in-procedural-macro/68204/2
+static FUNCTION_COUNTER: AtomicU32 = AtomicU32::new(0);
 
 pub(super) fn wasm_declare_init_fn(input: TokenStream) -> ParseResult<TokenStream> {
     if !input.is_empty() {
         return bail!(input, "macro expects no arguments");
     }
 
-    // Create sufficiently unique identifier without entire `uuid` (let alone `rand`) crate dependency.
-    let a = unsafe { libc::rand() };
-    let b = unsafe { libc::rand() };
+    let crate_name = env::var("CARGO_CRATE_NAME")
+        .expect("CARGO_CRATE_NAME env var not found. This macro must be run by Cargo.");
+
+    let crate_version = env::var("CARGO_PKG_VERSION")
+        .expect("CARGO_PKG_VERSION env var not found. This macro must be run by Cargo.")
+        .replace(['.', '+', '-'], "_"); // version must follow semver, which allows digits, dots, hyphens, and plus signs, and alphanumeric characters.
```

Comment:
This is a potentially ambiguous mapping:
```
1.2.3-alpha.4
1.2.3+alpha.4
```
both map to the same.

It's a very unlikely edge case, so I'm OK with the current implementation, but this should be noted in a comment.
Please also use full sentences in comments. :slightly_smiling_face: 

You can apply this verbatim:
```suggestion
        // SemVer version allows digits, alphanumerics, dots, hyphens and plus signs. Replacement may technically
        // map strings like "1.2.3-alpha.4" and "1.2.3+alpha.4" to the same identifier, but that's a very unlikely edge case.
        .replace(['.', '+', '-'], "_");
```